src/typescript-service.ts

```diff
@@ -338,6 +338,847 @@ export default class TypeScriptService {
 		});
 	}
 
+	getDocumentSymbol(params: DocumentSymbolParams): Promise<SymbolInformation[]> {
+		const uri = params.textDocument.uri;
+		return this.ensureFilesForHoverAndDefinition(uri).then(() => {
+			const fileName = util.uri2path(uri);
+			const config = this.projectManager.getConfiguration(uri);
+			return config.prepare(this.connection).then(() => {
+				this.projectManager.syncConfiguration(config, this.connection);
+				const sourceFile = config.program.getSourceFile(fileName);
+				const tree = config.service.getNavigationTree(fileName);
+				const limit = 100000;
+				const result: SymbolInformation[] = [];
+				this.flattenNavigationTreeItem(tree, null, sourceFile, result, limit);
+				return Promise.resolve(result);
+			});
+		});
+	}
+
+	getWorkspaceReference(params: util.WorkspaceReferenceParams): Promise<util.ReferenceInformation[]> {
+		const refInfo: util.ReferenceInformation[] = [];
+		return this.ensureFilesForWorkspaceSymbol().then(() => {
+			return Promise.all(this.projectManager.getConfigurations().map((config) => {
+				return config.prepare(this.connection).then((config) => {
+					this.projectManager.syncConfiguration(config, this.connection);
+					for (let source of config.service.getProgram().getSourceFiles()) {
+						if (source.fileName.indexOf("/node_modules/") !== -1) {
```

Comment:
For a safety I'd do:
`if (util.normalizePath(source.fileName).indexOf(`${path_.posix.sep}node_modules${path_.posix.sep}`) !== -1) `